packages/core/src/InlineEditor/InlineEditor.tsx

```diff
@@ -55,7 +56,13 @@ export interface HvInlineEditorProps
  * An Inline Editor allows the user to edit a record without making a major switch
  * between viewing and editing, making it an efficient method of updating a record.
  */
-export const HvInlineEditor = (props: HvInlineEditorProps) => {
+export const HvInlineEditor = fixedForwardRef(function HvInlineEditor<
+  C extends React.ElementType = typeof HvInput,
+>(
+  props: HvInlineEditorProps<C> &
```

Comment:
Can't we only do the following for `props`?

```suggestion
  props: HvInlineEditorProps<C>
```

Reply:
I'm not seeing how, if I remove this it will complain that types of props are incompatible, maybe I'm miss understanding something.

Reply:
And what if we do something like the following and leverage `PolymorphicComponentRef` like our other polymorphic components? 🙏 

```tsx
export type HvInlineEditorProps<C extends React.ElementType = typeof HvInput> =
  PolymorphicComponentRef<
    C,
    {
      /** The value of the form element. */
      value?: string;
      /** The default value of the form element. */
      defaultValue?: string;
      /** Whether the Edit icon should always be visible */
      showIcon?: boolean;
      /** Variant of the HvTypography to display */
      variant?: HvTypographyVariants;
      /** Called when the input is blurred. */
      onBlur?: (
        event: React.FocusEvent<HTMLTextAreaElement | HTMLInputElement>,
        value: string,
      ) => void;
      /** Called when the input value changes. */
      onChange?: (event: React.SyntheticEvent, value: string) => void;
      /** Props passed to the HvButton component */
      buttonProps?: HvButtonProps;
      /** Props passed to the HvTypography text component */
      typographyProps?: HvTypographyProps;
      /** Whether the editor is disabled or not. */
      disabled?: boolean;
      /** A Jss Object used to override or extend the styles applied to the empty state component. */
      classes?: HvInlineEditorClasses;
      /** The placeholder value of the input. */
      placeholder?: string;
    }
  >;

export const HvInlineEditor = fixedForwardRef(function HvInlineEditor<
  C extends React.ElementType = typeof HvInput,
>(props: HvInlineEditorProps<C>, ref: PolymorphicRef<C>) {
  const {
    className,
    classes: classesProp,
    value: valueProp,
    defaultValue = "",
    showIcon,
    component: InputComponent = HvInput,
    variant = "body",
    placeholder = "Enter text",
    onBlur,
    onChange,
    onKeyDown,
    buttonProps,
    typographyProps,
    disabled,
    ...others
  } = useDefaultProps("HvInlineEditor", props);

  (...)
});
```